backend/packages/Upgrade/src/api/services/ExperimentUserService.ts

```diff
@@ -33,32 +35,50 @@ export class ExperimentUserService {
     return this.userRepository.findOne({ id });
   }
 
-  public async create(users: Array<Partial<ExperimentUser>>, logger: UpgradeLogger): Promise<ExperimentUser[]> {
-    logger.info({ message: 'Create a new User. Metadata of the user =>', details: users });
-    // TODO: Pratik please review this eslint error, is this working as intended?
-    const multipleUsers = users.map((user) => {
-      // eslint-disable-next-line no-self-assign
-      user.id = user.id;
-      return user;
-    });
-    // insert or update in the database
-    const updatedUsers = await this.userRepository.save(multipleUsers);
+  public async upsertOnChange(
+    oldExperimentUser: RequestedExperimentUser,
+    newExperimentUser: Partial<ExperimentUser>,
+    logger: UpgradeLogger
+  ): Promise<ExperimentUser[]> {
+    if (
```

Comment:
In case, we have an experiment user without a group or working group for them the db call will always happen. We should avoid that. 
This if block should create a new user if it doesn't exist.

Reply:
code optimized